src/util/address.rs

```diff
@@ -344,6 +344,13 @@ impl From<WitnessVersion> for ::bech32::u5 {
     }
 }
 
+impl From<WitnessVersion> for ::bech32::Variant {
+    #[inline]
+    fn from(version: WitnessVersion) -> Self {
+        version.bech32_variant()
+    }
+}
+
```

Comment:
I don't like this commit. It seems to violate the spirit of `From`.

Reply:
Is it because you find those types unrelated? That's what I feel but curious if you have another reason.

Reply:
Yeah, I think that's what it is. It seems like this is changing, conceptually, what the data is.